src/commands/utility/PurgeInvitesCommand.js

```diff
@@ -0,0 +1,88 @@
+const Command = require('../Command');
+const util = require('../../util');
+const config = require('../../../config.json');
+
+class PurgeInvitesCommand extends Command {
+
+    static names = ['purgeinvites'];
+
+    static userPerms = ['MANAGE_GUILD'];
+
+    static botPerms = ['MANAGE_GUILD'];
+
+    static description = 'Purge invites based on age and maximum uses';
+
+    static usage = '[<minimum-age>] [<maximum-uses>]';
+
+    static comment = 'By default all invites older than 30 days with less than 10 uses are deleted.';
+
+    static supportsSlashCommands = true;
+
+    static private = true;
+
+    async execute() {
+        if (!config.featureWhitelist?.includes(this.source.getGuild().id)) {
+            return await this.sendError('This command is only allowed in whitelisted guilds');
+        }
+
+        const maxDate = Date.now() - (util.timeToSec(this.options.getString('minimum-age') ?? '30d') * 1000);
+        const maxUses = this.options.getInteger('max-uses') ?? 10;
+
+        let invites = await this.source.getGuild().invites.fetch({cache: false});
+        invites = invites.filter(invite => invite.createdAt < maxDate && maxUses >= invite.uses);
+
+        if (invites.size === 0) {
+            await this.reply('No invites matched your filters.');
+            return;
+        }
+
+        const {confirmed, component} = await this.getConfirmation(`Delete ${invites.size} invites older created before <t:${Math.floor(maxDate / 1000)}:d> with less than ${maxUses} uses!`);
+        if (!confirmed) {
+            if (component) {
+                await component.reply({
+                    ephemeral: true,
+                    content: 'No invites have been deleted'
+                });
+            }
+            else {
+                await this.reply('No invites have been deleted');
+            }
+            return;
+        }
+
+        await component.deferReply({ephemeral: true});
+        await component.editReply(`Deleting ${invites.size} invites...`);
+        await Promise.all(invites.map(invite => invite.delete()));
```

Comment:
I'm uncertain if it's a good idea to send delete requests for all those invites at the same time.